### DISCUSSION TITLE
Preserving the selection order in Multi-Select component

### DISCUSSION THREAD

**Discussion by sidx1024** [ariakit/ariakit#3646] [Questions] [answered]
Given a list of SelectItem with value `A, B, C, and D` if I select `B, A, C` the `value` will be `A, B, C`. Is there a way to preserve the order of selection?

**Comment by diegohaz** [comment, marked as answer]
If you use controlled state, you can sort the value yourself:

```jsx
const [value, setValue] = useState(["Apple", "Cake"]);
value.sort((a, b) => a.localeCompare(b));

<SelectProvider value={value} setValue={setValue}>
```

**Reply by sidx1024** [reply]
Oh okay, but I meant the order in which they were selected.

**Reply by diegohaz** [reply]
Hmm, then I don't understand your question. The default behavior is that selecting `B, A, C` will produce the value `B, A, C`.

**Reply by sidx1024** [reply]
Oh, my bad. I was mapping the results the wrong way. Thanks for your help!
